core/src/mindustry/game/MapObjectives.java

```diff
@@ -998,33 +1081,123 @@ public LineMarker(String text, float x1, float y1, float x2, float y2){
+                }
+            }
+        }
+    }
+
+    /** Displays a texture with specified name. */
+    public static class TextureMarker extends ObjectiveMarker{
+        public @TilePos Vec2 pos = new Vec2();
+        public float rotation = 0f, width = 0f, height = 0f; // Zero width/height scales marker to original texture's size
+        public String textureName = "";
+        public Color color = Color.white.cpy();
+
+        private transient TextureRegion fetchedRegion;
+
+        public TextureMarker(String textureName, float x, float y){
+            this.textureName = textureName;
+            this.pos.set(x, y);
+        }
+
+        public TextureMarker(){}
+
+        @Override
+        public void control(LMarkerControl type, double p1, double p2, double p3){
+            if(!Double.isNaN(p1)){
+                switch(type){
+                    case pos -> pos.x = (float)p1 * tilesize;
+                    case rotation -> rotation = (float)p1;
+                    case textureSize -> width = (float)p1 * tilesize;
+                    case color -> color.set(Tmp.c1.fromDouble(p1));
+                    default -> super.control(type, p1, p2, p3);
+                }
+            }
+
+            if(!Double.isNaN(p2)){
+                switch(type){
+                    case pos -> pos.y = (float)p2 * tilesize;
+                    case textureSize -> height = (float)p2 * tilesize;
+                    default -> super.control(type, p1, p2, p3);
+                }
             }
         }
 
         @Override
-        public void draw(){
-            if(hidden) return;
+        public void baseDraw(float x, float y, float scaleFactor){
+            if(textureName.isEmpty()) return;
 
-            if(outline){
-                Lines.stroke(stroke + 2f, Pal.gray);
-                Lines.line(pos1.x, pos1.y, pos2.x, pos2.y);
+            if(fetchedRegion == null) fetchedRegion = Core.atlas.find(textureName);
+
+            // Zero width/height scales marker to original texture's size
+            if(Mathf.equal(width, 0f)) width = fetchedRegion.width * fetchedRegion.scl() * Draw.xscl;
+            if(Mathf.equal(height, 0f)) height = fetchedRegion.height * fetchedRegion.scl() * Draw.yscl;
+
+            Draw.z(drawLayer);
+            if(fetchedRegion.found()){
+                Draw.color(color);
+                Draw.rect(fetchedRegion, x, y, width * scaleFactor, height * scaleFactor, rotation);
+            }else{
+                Draw.color(Color.white);
+                Draw.rect("error", x, y, width * scaleFactor, height * scaleFactor, rotation);
             }
+        }
+
+        @Override
+        public void drawWorld(){
+            baseDraw(pos.x, pos.y, autoscale ? 4f / renderer.getDisplayScale() : 1f);
+        }
 
-            Lines.stroke(stroke, color);
-            Lines.line(pos1.x, pos1.y, pos2.x, pos2.y);
+        @Override
+        public void drawMinimap(MinimapRenderer minimap){
```

Comment:
You have this same exact implementation for drawMinimap/drawWorld copy-pasted across most of the markers, maybe that should be the default implementation?

Reply:
> You have this same exact implementation for drawMinimap/drawWorld copy-pasted across most of the markers, maybe that should be the default implementation?

I've explained the problem about it:
> Some overriden methods code duplication is still there (due to differences in subclasses like not having position variable in the superclass)

And i don't really know how to solve this

Reply:
How about an abstract class `Vec2Marker` / `PointMarker` that have those position variables, and override the method?